tests/components/homekit_controller/test_config_flow.py

```diff
@@ -1,41 +1,55 @@
 """Tests for homekit_controller config flow."""
+from asyncio import Future
 import json
 from unittest import mock
 
-import homekit
+import aiohomekit
+from aiohomekit.model.characteristics import CharacteristicsTypes
+from aiohomekit.model.services import ServicesTypes
 import pytest
 
 from homeassistant.components.homekit_controller import config_flow
 from homeassistant.components.homekit_controller.const import KNOWN_DEVICES
 
 from tests.common import MockConfigEntry
-from tests.components.homekit_controller.common import (
-    Accessory,
-    FakeService,
-    setup_platform,
-)
+from tests.components.homekit_controller.common import Accessory, setup_platform
+
+
+def succeed(retval=None):
```

Comment:
Use `asynctest.CourotineMock` if you need to mock a coroutine function. `asynctest.patch` should be used when patching coroutine functions and normal functions.

Reply:
Weird, i thought about that when i was getting the PR ready - but i thought i the `succeed()` helper was already in there. Clearly not! Will get on this.

Reply:
OK i have replaced both of these helpers with a `CoroutineMock`. That test module didn't introduce any new patch sites so I didn't have to use `asynctest.patch` - but will keep that in mind for when i rework these tests to use `async_init` etc.

I'm also working on moving some of the testing support code upstream (the bits that aren't HA specific of course). Lots of the code in this diff will end up there, as it seems fragile to have the same mock functions defined 30+ times. But that seemed too big to bolt on to this PR as well.